src/main/java/uk/gov/hmcts/ccd/domain/model/definition/CaseField.java

```diff
@@ -140,6 +144,46 @@ public void propagateACLsToNestedFields() {
         propagateACLsToNestedFields(this, this.accessControlLists);
     }
 
+    @JsonIgnore
+    public FieldType getFieldTypeByPath(String path) {
+        if (StringUtils.isBlank(path)) {
+            return this.getFieldType();
+        } else {
+            CaseField caseFieldByPath = this.findNestedElementByPath(path);
+            return caseFieldByPath.getFieldType();
+        }
+    }
+
+    @JsonIgnore
+    public CaseField findNestedElementByPath(String path) {
+        if (StringUtils.isBlank(path)) {
+            throw new RuntimeException(format("Invalid blank element path for field %s.", this.id));
+        }
+        if (this.getFieldType().getChildren().isEmpty()) {
+            throw new RuntimeException(format("CaseField %s has no nested elements.", this.id));
```

Comment:
I would expect the endpoint to return a 400 Bad Request exception when this.getFieldType().getChildren().isEmpty(). Can we double check that a RuntimeException thrown here gets mapped by the endpoint to a 400 status code and not a 50X type of error?

Reply:
Returning BadRequestException that is our exception mapped to 400